src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryAccess.java

```diff
@@ -0,0 +1,744 @@
+package jdk.incubator.foreign;
+
+import java.lang.invoke.VarHandle;
+import java.nio.ByteOrder;
+
+/**
+ * This class defines ready-made static accessors which can be used to dereference memory segments in many ways.
+ * Each accessor (see {@link #getInt(MemoryAddress, long)} takes a <em>base</em> address and an offset (expressed in bytes).
+ * The final address at which the dereference will occur will be computed by offsetting the base address by
+ * the specified offset, as if by calling {@link MemoryAddress#addOffset(long)} on the specified base address.
+ */
+public final class MemoryAccess {
+
+    public MemoryAccess() {
+        // just the one
+    }
```

Comment:
Should be private I guess:
```suggestion
    private MemoryAccess() {
        // just the one
    }
```